rts/Map/ReadMap.cpp

```diff
@@ -679,18 +680,71 @@ void CReadMap::UpdateFaceNormals(const SRectangle& rect, bool initialize)
 	const int z2 = std::min(mapDims.mapym1, rect.z2 + 1);
 	const int x2 = std::min(mapDims.mapxm1, rect.x2 + 1);
 
-	for_mt_chunk(z1, z2 + 1, [&](const int y) {
-		float3 fnTL;
-		float3 fnBR;
-
-		for (int x = x1; x <= x2; x++) {
-			const int idxTL = (y    ) * mapDims.mapxp1 + x; // TL
-			const int idxBL = (y + 1) * mapDims.mapxp1 + x; // BL
+       for_mt_chunk(z1, z2 + 1, [&](const int y) {
+#ifdef USE_XSIMD
+               using batch = xsimd::batch<float, 8>;
```

Comment:
Can you please explain why you picked up `8` as a batch size?
We enforce / are forced to use SSE2, which has a native support for only `4` floats.

Reply:
I think I used to use something like this:
```
       using SIMDVfloat = xsimd::simd_type<float>;

       // size is adjusted based on SIMD extensions supported
       // SIMD extensions are conditionally enabled based on compiler flags
       constexpr std::size_t simdSize = SIMDVfloat::size;
```
Maybe `xsimd` semantics got changed since then. But overall you're on the right track: don't hardcode the SIMD factor, let the compiler magic figure it out.